### PR TITLE
Python HTTPTrigger Lesson

### PR DESCRIPTION
**Pull request by Dzvezdana**
I added:
- Python directory in src containing 4 Python lessons following the TypeScript template: basic HTTP Function App, changed the template for a GET request,  changed the template for a POST request, and a custom "type-safe" function
- homework (NLP application)
- lesson markdowns for the HTTP examples, prerequisite, deployment, as well as the homework
- project to the workspaces folder
- tests

~~Only a tour is missing, but I'll add it after I gather some initial feedback on this PR.~~
Added tour files for the Python lessons and homework. I fixed some broken tour paths in the typescript project.

### CODE REVIEW

File: lessons/python/prerequisites/prerequisites-ts.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Prerequisites (Python)
```
Comment (marcduiker) [inline lessons/python/prerequisites/prerequisites-ts.md]: Could you rename the file to prerequisites-python.md ?

File: lessons/python/prerequisites/prerequisites-ts.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+|[VSCode Azure Functions extension](https://github.com/Microsoft/vscode-azurefunctions)|All|Extension for VSCode to easily develop and manage Azure Functions.
+|[Azure Functions Core Tools](https://github.com/Azure/azure-functions-core-tools)|All|Azure Functions runtime and CLI for local development.
+|[RESTClient for VSCode](https://marketplace.visualstudio.com/items?itemName=humao.rest-client) or [Postman](https://www.postman.com/)|All|An extension or  application to make HTTP requests.
+|[Azure CLI](https://docs.microsoft.com/en-us/cli/azure/what-is-azure-cli)|Deployment|Command line interface used to manage Azure resources. Can be run on your local dev environment, in a deployment pipeline or in the [Azure Cloud Shell](https://docs.microsoft.com/en-us/azure/cloud-shell/overview).
+
+> 📝 **Tip** - Azure Functions only support the long term support (LTS) versions of Node.js as shown in the [Azure Functions runtime versions overview](https://docs.microsoft.com/en-us/azure/azure-functions/functions-versions). Consequently, there is always a bit of a lag between the official availability of a Node.js LTS release and the official support of this release in Azure Functions.
```
Comment (marcduiker) [inline lessons/python/prerequisites/prerequisites-ts.md]: This section can be adjusted to mention Python versions. I suggest adding this link (https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-python?tabs=azurecli-linux%2Capplication-level#python-version) so we point to the official docs instead of repeating content that will get outdated quickly

File: lessons/python/prerequisites/prerequisites-ts.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    ```cmd
+    C:\dev\azfuncuni\mkdir src
+    C:\dev\azfuncuni\mkdir tst
+    ```
+
+You should be good to go now!
+
+---
+[🔼 Lessons Index](../../README.md) | [Next lesson ▶](../http/http-lesson-ts.md)
```
Comment (marcduiker) [inline lessons/python/prerequisites/prerequisites-ts.md]:
```suggestion
[🔼 Lessons Index](../../README.md) | [Next lesson ▶](../http/http-lesson-python.md)
```

File: .vscode/extensions.json [hunk 4]
```diff
@@ -1,6 +1,7 @@
 {
   "recommendations": [
     "ms-azuretools.vscode-azurefunctions",
+    "ms-python.python",
```
Comment (marcduiker) [inline .vscode/extensions.json:4]: Could you create a new `.vscode` folder with these 4 files scoped for Python only inside the `src/python/AzureFunction.Http folder`? When a student then opens the workspace for this lesson, all of the settings and run commands will still work.

I will clean up this highest level `.vscode` folder to remove the lesson and language specific parts.

**Review by marcduiker — CHANGES_REQUESTED** [review]
It looks really good already! I only have some minor comments and things can need to be moved.

File: lessons/README.md [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
-|HTTP Trigger|[.NET Core](dotnetcore31/http/http-lesson-dotnet.md)|-|[TypeScript](typescript/http/http-lesson-ts.md)|[PowerShell (VS Code)](PowerShell/http/http-lesson-powershell.md), [PowerShell (portal)](PowerShell/http/http-lesson-powershell-portal.md)|-|Marc,Gwyneth, Barbara, Caroline, Christian
-|Blob Trigger & Bindings|[.NET Core](dotnetcore31/blob/blob-lesson-dotnet.md)|-|[TypeScript](typescript/blob/blob-lesson-ts.md)|-|-|Marc, Gwyneth, Christian
-|Queue Trigger & Bindings|[.NET Core](dotnetcore31/queue/queue-lesson-dotnet.md)|-|-|-|-|Marc
-|Table Bindings|[.NET Core](dotnetcore31/table/table-lesson-dotnet.md)|-|-|-|-|Marc
-|Deployment to Azure|-|-|-|-|[Agnostic](deployment/deployment-lesson.md)|Marc
+Lesson|.NET Core|.NET 6|Typescript|PowerShell|Python|Agnostic|Contributions by
+|-|-|-|-|-|-|-|-
+|General Prerequisites|[.NET Core](dotnetcore31/prerequisites/prerequisites-dotnet.md)|-|[TypeScript](typescript/prerequisites/prerequisites-ts.md)|[PowerShell](PowerShell/prerequisites/prerequisites-powershell.md)|[Python](python/prerequisites/prerequisites-ts.md)|-|Marc, Gwyneth, Barbara, Christian, Dana
```
Comment (marcduiker) [inline lessons/README.md:5]:
```suggestion
|General Prerequisites|[.NET Core](dotnetcore31/prerequisites/prerequisites-dotnet.md)|-|[TypeScript](typescript/prerequisites/prerequisites-ts.md)|[PowerShell](PowerShell/prerequisites/prerequisites-powershell.md)|[Python](python/prerequisites/prerequisites-python.md)|-|Marc, Gwyneth, Barbara, Christian, Dana
```

File: lessons/python/http/http-lesson-python.md [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+## 0. Prerequisites
+
+| Prerequisite | Exercise
+| - | -
+| An empty local folder / git repo | 1-5
+| Azure Functions Core Tools | 1-5
+| VS Code with Azure Functions extension| 1-5
+| Rest Client for VS Code or Postman | 1-5
+
+See [Python prerequisites](../prerequisites/prerequisites-ts.md) for more details.
```
Comment (marcduiker) [inline lessons/python/http/http-lesson-python.md:32]:
```suggestion
See [Python prerequisites](../prerequisites/prerequisites-python.md) for more details.
```

**Review by marcduiker — COMMENTED** [review]
Two small file renames so the links keep working.

File: .vscode/extensions.json (hunk 4 above)
Comment (Dzvezdana) [reply .vscode/extensions.json]: I moved the `vscode` folder to `src/python/AzureFunctions.Http`, and I reverted the changes in the root `vscode`. Hope it looks okay now! :)

File: .tours/typescript/http/.tours/01creating-a-function-app.tour [hunk 7]
```diff
@@ -3,7 +3,7 @@
   "title": "TypeScript HTTP - 01. Creating a Function App",
   "steps": [
     {
-      "file": "../../../src/typescript/AzureFunctions.Http/01CreatingAFunctionApp/HelloWorldHttpTrigger/index.ts",
```
Comment (marcduiker) [inline .tours/typescript/http/.tours/01creating-a-function-app.tour:6]: Could you revert this change? The tour files are supposed to be used from within a Workspace, that's why the paths is so long. Unfortunately this does mean CodeTours can't be used in the folder itself :/.

File: .tours/python/http/.tours/01creating-a-function-app.tour [hunk 8]
```diff
@@ -0,0 +1,201 @@
+{
+  "$schema": "https://aka.ms/codetour-schema",
+  "title": "Python HTTP - 01. Creating a Function App",
+  "steps": [
+    {
+      "file": "src/python/AzureFunctions.Http/01CreatingAFunctionApp/HelloWorldHttpTrigger/__init__.py",
```
Comment (marcduiker) [inline .tours/python/http/.tours/01creating-a-function-app.tour:6]: Please make sure you test everything when you opened the Python HTTP lesson workspace.  The tour files are supposed to be used from within a Workspace, and not from the root folder itself.

File: src/python/AzureFunctions.Http/.vscode/launch.json [hunk 9]
```diff
@@ -0,0 +1,18 @@
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "Attach to .NET Functions",
```
Comment (marcduiker) [inline src/python/AzureFunctions.Http/.vscode/launch.json:5]: You can remove the .NET configuration

File: src/python/AzureFunctions.Http/.vscode/tasks.json [hunk 10]
```diff
@@ -0,0 +1,428 @@
+{
+	"version": "2.0.0",
+	"tasks": [
+		{
+			"label": "clean http",
```
Comment (marcduiker) [inline src/python/AzureFunctions.Http/.vscode/tasks.json:5]: You can remove all the .NET related tasks from his one.

**Review by marcduiker — CHANGES_REQUESTED** [review]
Some small things to clean up and to make sure CodeTour works from the Python HTTP lesson workspace.

File: .tours/typescript/http/.tours/01creating-a-function-app.tour (hunk 7 above)
Comment (Dzvezdana) [reply .tours/typescript/http/.tours/01creating-a-function-app.tour]: I reverted the change, but this means the Typescript tour now works only for some steps. All steps that don't use 
"../../../src" wont work when I open the Typescript workspace. For example, I can't run tours 2-4 and steps 10 to the end in Tour 1. Might be a Mac thing. Shall I prefix all paths in the Typescript tour with "../../../src"?

Comment (marcduiker) [reply .tours/typescript/http/.tours/01creating-a-function-app.tour]: Ah ok, yes that's probably an OS specific thing. I'll fix that in a separate PR :).

**Review by marcduiker — APPROVED** [review]
LGTM
